File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1045_basico.js
import { question } from "readline-sync";

function main() {
    const valores = get_number();

    let Nvalores = valores.split(` `);
    let A = parseFloat(Nvalores[0]);
    let B = parseFloat(Nvalores[1]);
    let C = parseFloat(Nvalores[2]);

    const ordem = Ordem_Decrescente(A, B, C);

    if (!Eh_Triangulo(ordem)) {
        if (Eh_Obtusangulo(ordem)) {
            print(`TRIANGULO OBTUSANGULO`);
            if (Eh_Equilatero(ordem)) {
                print(`TRIANGULO EQUILATERO`);
            } else if (Eh_Isosceles(ordem)) {
                print(`TRIANGULO ISOSCELES`)
            }
        } else if (Eh_Acutangulo(ordem)) {
            print(`TRIANGULO ACUTANGULO`);
            if (Eh_Equilatero(ordem)) {
                print(`TRIANGULO EQUILATERO`);
            } else if (Eh_Isosceles(ordem)) {
                print(`TRIANGULO ISOSCELES`);
            }
        } else if (Eh_Retangulo(ordem)) {
            print(`TRIANGULO RETANGULO`);
        }
    } else {
        print(`NAO FORMA TRIANGULO`);
    }
}

function get_number() {
    return question();
}

function print(texto) {
    return console.log(texto);
}

function Ordem_Decrescente(num1, num2, num3) {
    let decrescente;

    if (num1 >= num2 && num1 >= num3) {
        if (num2 >= num3) {
            decrescente = [num1, num2, num3];
        } else {
            decrescente = [num1, num3, num2];
        }
    } else if (num2 >= num1 && num2 >= num3) {
        if (num1 >= num3) {
            decrescente = [num2, num1, num3];
        } else {
            decrescente = [num2, num3, num1];
        }
    } else if (num3 >= num1 && num3 >= num2) {
        if (num2 >= num1) {
            decrescente = [num3, num2, num1];
        } else {
            decrescente = [num3, num1, num2];
        }
    } else {

    }
    return decrescente;
}

function Eh_Triangulo(lista) {
    return (lista[0] >= lista[1] + lista[2]);
}

function Eh_Obtusangulo(lista) {
    return (lista[0] ** 2 > lista[1] ** 2 + lista[2] ** 2);
}

function Eh_Isosceles(lista) {
    return (lista[0] === lista[1] || lista[0] === lista[2] ||
        lista[1] === lista[2]);
}

function Eh_Equilatero(lista) {
    return (lista[0] === lista[1] && lista[0] === lista[2]);
}

function Eh_Acutangulo(lista) {
    return (lista[0] ** 2 < lista[1] ** 2 + lista[2] ** 2);
}

function Eh_Retangulo(lista) {
    return (lista[0] ** 2 === lista[1] ** 2 + lista[2] ** 2);
}

main();

File: Exercicios_ED_I/Bee/1045_basico.cpp
#include <stdio.h>
#include <iostream>
#include <cmath>
using namespace std;

void ordem_Decrescente(double &a, double &b, double &c) {
    if (a < b) swap(a, b);
    if (a < c) swap(a, c);
    if (b < c) swap(b, c);
}

main() {
    double a, b, c;
    cin >> a >> b >> c;
    ordem_Decrescente(a, b, c);

    if (a >= b + c) {
        cout << "NAO FORMA TRIANGULO" << endl;
    } else if (pow(a, 2) == pow(b, 2) + pow(c, 2)) {
        cout << "TRIANGULO RETANGULO" << endl;
    } else if (pow(a, 2) > pow(b, 2) + pow(c, 2)) {
        cout << "TRIANGULO OBTUSANGULO" << endl;
    } else if (pow(a, 2) < pow(b, 2) + pow(c, 2)) {
        cout << "TRIANGULO ACUTANGULO" << endl;
    }

    if(a == b && b == c) {
        cout << "TRIANGULO EQUILATERO" << endl;
    } else if (a == b || a == c || b == c) {
        cout << "TRIANGULO ISOSCELES" << endl;
    }
    
}